src/traces/indicator/plot.js

```diff
@@ -0,0 +1,788 @@
+
+var d3 = require('d3');
+
+var Lib = require('../../lib');
+var rad2deg = Lib.rad2deg;
+var MID_SHIFT = require('../../constants/alignment').MID_SHIFT;
+var Drawing = require('../../components/drawing');
+var cn = require('./constants');
+var svgTextUtils = require('../../lib/svg_text_utils');
+
+var Axes = require('../../plots/cartesian/axes');
+var handleAxisDefaults = require('../../plots/cartesian/axis_defaults');
+var handleAxisPositionDefaults = require('../../plots/cartesian/position_defaults');
+var axisLayoutAttrs = require('../../plots/cartesian/layout_attributes');
+
+var anchor = {
+    'left': 'start',
+    'center': 'middle',
+    'right': 'end'
+};
+var position = {
+    'left': 0,
+    'center': 0.5,
+    'right': 1
+};
+
+module.exports = function plot(gd, cdModule, transitionOpts, makeOnCompleteCallback) {
+    var fullLayout = gd._fullLayout;
+    var onComplete;
+
+    // If transition config is provided, then it is only a partial replot and traces not
+    // updated are removed.
+    var hasTransition = transitionOpts && transitionOpts.duration > 0;
+
+    if(hasTransition) {
+        if(makeOnCompleteCallback) {
+            // If it was passed a callback to register completion, make a callback. If
+            // this is created, then it must be executed on completion, otherwise the
+            // pos-transition redraw will not execute:
+            onComplete = makeOnCompleteCallback();
+        }
+    }
+
+    Lib.makeTraceGroups(fullLayout._indicatorlayer, cdModule, 'trace').each(function(cd) {
+        var cd0 = cd[0];
+        var trace = cd0.trace;
+        var plotGroup = d3.select(this);
+
+        // Elements in trace
+        var hasGauge = trace._hasGauge;
+        var isAngular = trace._isAngular;
+        var isBullet = trace._isBullet;
+
+        // Domain size
+        var domain = trace.domain;
+        var size = {
+            w: fullLayout._size.w * (domain.x[1] - domain.x[0]),
+            h: fullLayout._size.h * (domain.y[1] - domain.y[0]),
+            l: fullLayout._size.l + fullLayout._size.w * domain.x[0],
+            r: fullLayout._size.r + fullLayout._size.w * (1 - domain.x[1]),
+            t: fullLayout._size.t + fullLayout._size.h * (1 - domain.y[1]),
+            b: fullLayout._size.b + fullLayout._size.h * (domain.y[0])
+        };
+        var centerX = size.l + size.w / 2;
+        var centerY = size.t + size.h / 2;
+
+
+        // Angular gauge size
+        var radius = Math.min(size.w / 2, size.h); // fill domain
+        var innerRadius = cn.innerRadius * radius;
+
+        // Position numbers based on mode and set the scaling logic
+        var numbersX, numbersY, numbersScaler;
+        var numbersAlign = trace.align || 'center';
+
+        numbersY = centerY;
+        if(!hasGauge) {
+            numbersX = size.l + position[numbersAlign] * size.w;
+            numbersScaler = function(el) {
+                return fitTextInsideBox(el, 0.9 * size.w, 0.9 * size.h);
```

Comment:
I'm curious why you picked `0.9` as factor here? How bad do things look if we replace it with `1`?

Reply:
Having `0.9` for angular gauge makes sense as we want some sort of padding inside the arc, but for gauge-less `indicator` traces, it might be best to try to fit the `domain` box.

Reply:
Now fit the domain box in e6d8525ab07c56ca10bdbac6762b06d1d57bf830